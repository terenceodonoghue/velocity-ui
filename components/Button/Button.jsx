/** @jsx jsx */
import { jsx } from '@emotion/core';
import * as S from './Button.styles';

const Contained = props => <S.Contained {...props} />;
const Outlined = props => <S.Outlined {...props} />;
const Text = props => <S.Text {...props} />;
const Fab = props => <S.Fab {...props} />;

export default {
  Contained,
  Outlined,
  Text,
  Fab,
};
