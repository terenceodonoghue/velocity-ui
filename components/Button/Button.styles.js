import { css } from '@emotion/core';
import styled from '@emotion/styled';

const base = css`
  display: flex;
  align-items: center;
  justify-content: center;
  border: none;
  padding: 0;
  font-size: var(--base-font-size);
  font-weight: var(--font-weight-medium);
  cursor: pointer;
  transition: var(--transition);

  &:disabled {
    opacity: 0.3;
    pointer-events: none;
  }

  &:focus {
    outline: 0;
  }
`;

export const Contained = styled.button`
  ${base};
  border-radius: 4px;
  height: 40px;
  width: ${props => props.width || '200px'};
  background-color: var(--color-clear-blue);
  color: white;

  &:enabled:hover {
    background-color: var(--color-clear-blue-dark);
  }
`;

export const Outlined = styled.button`
  ${base};
  border-radius: 4px;
  height: 40px;
  width: ${props => props.width || '200px'};
  background: rgba(46, 91, 255, 0.2);
  color: var(--color-clear-blue);

  &:enabled:hover {
    background-color: rgba(46, 91, 255, 0.3);
  }
`;

export const Text = styled.button`
  ${base};
  border-radius: 4px;
  height: 40px;
  width: ${props => props.width || '200px'};
  background: none;
  color: var(--color-clear-blue);

  &:enabled:hover {
    background-color: rgba(46, 91, 255, 0.1);
  }
`;

export const Fab = styled.button`
  ${base};
  border-radius: 50%;
  height: 70px;
  width: 70px;
  background-color: var(--color-clear-blue);
  color: white;

  &:enabled:hover {
    background-color: var(--color-clear-blue-dark);
  }
`;
