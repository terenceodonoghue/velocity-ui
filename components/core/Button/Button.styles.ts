import { css, FunctionInterpolation, Theme } from '@emotion/react';
import { darken } from 'polished';

export const button: FunctionInterpolation<Theme> = ({
  transitions,
  typography,
}) => css`
  display: flex;
  align-items: center;
  justify-content: center;
  border: none;
  border-radius: 4px;
  height: 40px;
  padding: 0;
  width: 200px;
  font-weight: ${typography.fontWeight.medium};
  cursor: pointer;
  transition: background-color ${transitions.duration.standard}ms
    ${transitions.easing.easeInOut};

  &:disabled {
    opacity: 0.3;
    pointer-events: none;
  }

  &:focus {
    outline: 0;
  }
`;

export const contained: FunctionInterpolation<Theme> = ({ palette }) => css`
  background-color: ${palette.accent};
  color: ${palette.neutral[50]};

  &:enabled:hover {
    background-color: ${darken(palette.tonalOffset, palette.accent)};
  }
`;

export const outlined: FunctionInterpolation<Theme> = ({ palette }) => css`
  background: ${palette.primary};
  color: ${palette.accent};

  &:enabled:hover {
    background-color: ${darken(palette.tonalOffset, palette.accent)};
  }
`;

export const text: FunctionInterpolation<Theme> = ({ palette }) => css`
  background: none;
  color: ${palette.accent};

  &:enabled:hover {
    background-color: ${darken(palette.tonalOffset, palette.accent)};
  }
`;
