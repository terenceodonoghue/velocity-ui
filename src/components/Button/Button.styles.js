import { css } from '@emotion/core';
import styled from '@emotion/styled';
import Global from '../Global/Global';

const {
  variables: { colors, effects, typography },
} = Global;

const base = css`
  display: flex;
  align-items: center;
  justify-content: center;
  border: none;
  padding: 0;
  font-size: ${typography.baseFontSize};
  font-weight: ${typography.fontWeightMedium};
  cursor: pointer;
  transition: ${effects.transition};

  &:disabled {
    opacity: 0.3;
    pointer-events: none;
  }

  &:focus {
    outline: 0;
  }
`;

export const Contained = styled.button`
  ${base};
  border-radius: 4px;
  height: 40px;
  width: 200px;
  background-color: ${colors.clearBlue};
  color: ${colors.white};

  &:enabled:hover {
    background-color: ${colors.clearBlueDark};
  }
`;

export const Outlined = styled.button`
  ${base};
  border-radius: 4px;
  height: 40px;
  width: 200px;
  background: rgba(46, 91, 255, 0.2);
  color: ${colors.clearBlue};

  &:enabled:hover {
    background-color: rgba(46, 91, 255, 0.3);
  }
`;

export const Text = styled.button`
  ${base};
  border-radius: 4px;
  height: 40px;
  width: 200px;
  background: none;
  color: ${colors.clearBlue};

  &:enabled:hover {
    background-color: rgba(46, 91, 255, 0.1);
  }
`;

export const Fab = styled.button`
  ${base};
  border-radius: 50%;
  height: 70px;
  width: 70px;
  background-color: ${colors.clearBlue};
  color: ${colors.white};

  &:enabled:hover {
    background-color: ${colors.clearBlueDark};
  }
`;
