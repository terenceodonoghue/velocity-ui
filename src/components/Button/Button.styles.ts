import { css, InterpolationWithTheme } from '@emotion/core';
import { Theme } from '..';

export const button: InterpolationWithTheme<Theme> = (theme) => css`
  display: flex;
  align-items: center;
  justify-content: center;
  border: none;
  padding: 0;
  font-size: ${theme.typography.baseFontSize};
  font-weight: ${theme.typography.fontWeightMedium};
  cursor: pointer;
  transition: ${theme.effects.transition};

  &:disabled {
    opacity: 0.3;
    pointer-events: none;
  }

  &:focus {
    outline: 0;
  }
`;

export const contained: InterpolationWithTheme<Theme> = (theme) => css`
  ${button(theme)}
  border-radius: 4px;
  height: 40px;
  width: 200px;
  background-color: ${theme.colors.blue};
  color: ${theme.colors.white};

  &:enabled:hover {
    background-color: ${theme.colors.blueDark};
  }
`;

export const icon: InterpolationWithTheme<Theme> = (theme) => css`
  ${button(theme)}
  background: none;
  border-radius: 5px;
  padding: 8px;
  &:enabled:hover {
    background-color: rgba(46, 91, 255, 0.1);
  }
`;

export const outlined: InterpolationWithTheme<Theme> = (theme) => css`
  ${button(theme)}
  border-radius: 4px;
  height: 40px;
  width: 200px;
  background: rgba(46, 91, 255, 0.2);
  color: ${theme.colors.blue};

  &:enabled:hover {
    background-color: rgba(46, 91, 255, 0.3);
  }
`;

export const text: InterpolationWithTheme<Theme> = (theme) => css`
  ${button(theme)}
  border-radius: 4px;
  height: 40px;
  width: 200px;
  background: none;
  color: ${theme.colors.blue};

  &:enabled:hover {
    background-color: rgba(46, 91, 255, 0.1);
  }
`;

export const fab: InterpolationWithTheme<Theme> = (theme) => css`
  ${button(theme)}
  border-radius: 50%;
  height: 70px;
  width: 70px;
  background-color: ${theme.colors.blue};
  color: ${theme.colors.white};

  &:enabled:hover {
    background-color: ${theme.colors.blueDark};
  }
`;
