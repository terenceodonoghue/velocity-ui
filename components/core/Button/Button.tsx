import React, { forwardRef } from 'react';
import * as css from './Button.styles';
import type { ButtonProps } from './Button.types';

const Button = forwardRef<HTMLButtonElement, ButtonProps>(
  ({ variant = 'contained', ...props }, ref) => (
    <button
      css={[css.button, css[variant]]}
      data-testid="button"
      ref={ref}
      type="button"
      {...props}
    />
  ),
);

export default Button;
