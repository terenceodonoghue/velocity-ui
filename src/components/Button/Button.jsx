import React from 'react';
import * as S from './Button.styles';

export const Contained = (props) => <S.Contained {...props} />;

export const Outlined = (props) => <S.Outlined {...props} />;

export const Text = (props) => <S.Text {...props} />;

export const Fab = (props) => <S.Fab {...props} />;
