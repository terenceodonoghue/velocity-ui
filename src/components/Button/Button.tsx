/** @jsx jsx */
import { jsx } from '@emotion/core';
import { ButtonHTMLAttributes, FunctionComponent } from 'react';
import * as css from './Button.styles';

export const Contained: FunctionComponent<ButtonHTMLAttributes<
  HTMLButtonElement
>> = (props) => <button css={css.contained} type="button" {...props} />;

export const Outlined: FunctionComponent<ButtonHTMLAttributes<
  HTMLButtonElement
>> = (props) => <button css={css.outlined} type="button" {...props} />;

export const Text: FunctionComponent<ButtonHTMLAttributes<
  HTMLButtonElement
>> = (props) => <button css={css.text} type="button" {...props} />;

export const Fab: FunctionComponent<ButtonHTMLAttributes<HTMLButtonElement>> = (
  props,
) => <button css={css.fab} type="button" {...props} />;
