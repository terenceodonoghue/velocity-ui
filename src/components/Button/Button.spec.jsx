import React from 'react';
import Button from './Button';

describe('Button', () => {
  describe('Contained', () => {
    it('matches snapshot', () => {
      const tree = renderer
        .create(<Button.Contained>Button Text</Button.Contained>)
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('Outlined', () => {
    it('matches snapshot', () => {
      const tree = renderer
        .create(<Button.Outlined>Button Text</Button.Outlined>)
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('Text', () => {
    it('matches snapshot', () => {
      const tree = renderer
        .create(<Button.Text>Button Text</Button.Text>)
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('Fab', () => {
    it('matches snapshot', () => {
      const tree = renderer.create(<Button.Fab>+</Button.Fab>).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
