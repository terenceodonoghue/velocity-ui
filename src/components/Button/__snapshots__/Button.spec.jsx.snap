// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Contained matches snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: none;
  padding: 0;
  font-size: var(--base-font-size);
  font-weight: var(--font-weight-medium);
  cursor: pointer;
  -webkit-transition: var(--transition);
  transition: var(--transition);
  border-radius: 4px;
  height: 40px;
  width: 200px;
  background-color: var(--color-clear-blue);
  color: white;
}

.emotion-0:disabled {
  opacity: 0.3;
  pointer-events: none;
}

.emotion-0:focus {
  outline: 0;
}

.emotion-0:enabled:hover {
  background-color: var(--color-clear-blue-dark);
}

<button
  className="emotion-0 emotion-1"
>
  Button Text
</button>
`;

exports[`Button Fab matches snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: none;
  padding: 0;
  font-size: var(--base-font-size);
  font-weight: var(--font-weight-medium);
  cursor: pointer;
  -webkit-transition: var(--transition);
  transition: var(--transition);
  border-radius: 50%;
  height: 70px;
  width: 70px;
  background-color: var(--color-clear-blue);
  color: white;
}

.emotion-0:disabled {
  opacity: 0.3;
  pointer-events: none;
}

.emotion-0:focus {
  outline: 0;
}

.emotion-0:enabled:hover {
  background-color: var(--color-clear-blue-dark);
}

<button
  className="emotion-0 emotion-1"
>
  +
</button>
`;

exports[`Button Outlined matches snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: none;
  padding: 0;
  font-size: var(--base-font-size);
  font-weight: var(--font-weight-medium);
  cursor: pointer;
  -webkit-transition: var(--transition);
  transition: var(--transition);
  border-radius: 4px;
  height: 40px;
  width: 200px;
  background: rgba(46,91,255,0.2);
  color: var(--color-clear-blue);
}

.emotion-0:disabled {
  opacity: 0.3;
  pointer-events: none;
}

.emotion-0:focus {
  outline: 0;
}

.emotion-0:enabled:hover {
  background-color: rgba(46,91,255,0.3);
}

<button
  className="emotion-0 emotion-1"
>
  Button Text
</button>
`;

exports[`Button Text matches snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: none;
  padding: 0;
  font-size: var(--base-font-size);
  font-weight: var(--font-weight-medium);
  cursor: pointer;
  -webkit-transition: var(--transition);
  transition: var(--transition);
  border-radius: 4px;
  height: 40px;
  width: 200px;
  background: none;
  color: var(--color-clear-blue);
}

.emotion-0:disabled {
  opacity: 0.3;
  pointer-events: none;
}

.emotion-0:focus {
  outline: 0;
}

.emotion-0:enabled:hover {
  background-color: rgba(46,91,255,0.1);
}

<button
  className="emotion-0 emotion-1"
>
  Button Text
</button>
`;
